### ISSUE TITLE
Add documentation for creating and uploading embeddings through the pip package (non cli)

### ISSUE THREAD

**Issue by IgorSusmelj** [lightly-ai/lightly#132]
The CLI way of obtaining and uploading embeddings is described. It's pretty minimal but works. However, there is no information on how to do this yourself without using the CLI. 

I would prefer steps starting from having a pre-trained model.
```python
class Net(nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        self.conv1 = nn.Conv2d(1, 32, 3, 1)
        self.conv2 = nn.Conv2d(32, 64, 3, 1)
        self.dropout1 = nn.Dropout(0.25)
        self.dropout2 = nn.Dropout(0.5)
        self.fc1 = nn.Linear(9216, 128)
        self.fc2 = nn.Linear(128, 10)

    def embed(self, x):
        x = self.conv1(x)
        x = F.relu(x)
        x = self.conv2(x)
        x = F.relu(x)
        x = F.max_pool2d(x, 2)
        x = self.dropout1(x)
        x = torch.flatten(x, 1)
        x = self.fc1(x)
        x = F.relu(x)
        x = self.dropout2(x)
        x = self.fc2(x)
        return x

    def forward(self, x):
        x = self.embed(x)
        output = F.log_softmax(x, dim=1)
        return output

model = Net()

# let's assume here model is pre-trained
# I can get the embeddings and predictions using
embedding = model.embed(x)
prediction = model(x)

# TODO: how to create embeddings and upload them to lightly
```

**Comment by MalteEbner** [comment]
The code for uploading the embeddings from an embedding.csv is pretty easy:

```python
api_workflow_client = ApiWorkflowClient(token=YOUR_TOKEN, dataset_id=YOUR_DATASET_ID)
api_workflow_client.upload_embeddings(name="embedding-1", path_to_embeddings_csv=path_to_embeddings_csv)
```

**Comment by philippmwirth** [comment]
You also need the code to create the CSV file before you can upload. The question for me is where to put it in the docs? Once we have a place it's easy to write down :)

**Comment by MalteEbner** [comment]
I also found the autodoc of the upload of the embeddings in https://github.com/lightly-ai/lightly/blob/develop/docs/source/lightly.api.rst. Could we add a better description there?

**Comment by philippmwirth** [comment]
This is outdated. The new autodocs should cover the `ApiWorkflowClient` way of uploading embeddings.

**Comment by philippmwirth** [comment]
The question here is whether we really want to add a whole section on how to get the embeddings from a custom pre-trained model. IMO most users will be able to do that and we should focus on showing them how to store them in a csv file and then upload them. So my proposition is something like this:


If you have a numpy array of image embeddings, the filenames of the images, and categorical pseudo-labels, you can use the [`save_embeddings`](https://docs.lightly.ai/lightly.utils.html#lightly.utils.io.save_embeddings) function to store them in a lightly-compatible CSV format and upload them from your Python code or using the CLI. The following snippet shows how to upload the embeddings from Python:

```python
from lightly.utils import save_embeddings
from lightly.api.api_workflow_client import ApiWorkflowClient

# save embeddings
save_embeddings('embeddings.csv', embeddings, labels, filenames)

# upload them to the platform
client = ApiWorkflowClient(token='123', dataset_id='xyz')
client.upload_embeddings('embeddings.csv', name='my-embeddings')
```
